sdk/core/core-amqp/src/util/cancellableDelay.ts

```diff
@@ -0,0 +1,17 @@
+import { AbortSignalLike, AbortError } from "@azure/abort-controller";
+
+export function cancellableDelay(delayInMs: number, abortSignal?: AbortSignalLike): Promise<void> {
```

Comment:
Any reason to have this in a separate file instead of adding this to the existing utils.ts file where the simple `delay` lives?

Reply:
Also, when no abortsignal is provided, this function is the same as `delay` correct?
If so, then why not update the existing `delay` function instead of creating a new one?